### PR TITLE
zero-noexert-exhaustion bugfix script, doc, and control-panel registry entry

### PR DESCRIPTION
**Pull request by terribleperson**
zero-noexert-exhaustion and its accompanying control-panel registry entry are my attempt at a workaround for DF bug 8389. NOEXERT-tagged creatures ignore most sources of Exhaustion, but are unable to reduce their Exhaustion. 

Individual Combat Drill (henceforth referred to as ICD) seems to raise the Exhaustion of a NOEXERT Intelligent Undead dwarf by approximately 50 Exhaustion every 9 ticks. At 2000 Exhaustion, a dwarf becomes Tired, and at 3000 Exhaustion the ICD activity becomes Individual Combat Drill/Resting. 

zero-noexert-exhaustion loops through every unit in df.global.world.units.active, checks if they're NOEXERT using similar logic to the OPPOSED_TO_LIFE check, and sets the Exhaustion of any NOEXERT units to 0. Running the script every 350 ticks should hopefully prevent NOEXERT-tagged dwarves (Necromancers, Vampires, and Intelligent Undead) from becoming tired and allow them to participate in military training more normally. This is the frequency the control-panel registry entry uses.

It's possible this workaround will have knock-on effects since most NOEXERT dwarves have far fewer distractions than a normal dwarf, but normal dwarves are already nearly immune to the Exhaustion effects of ICD. The scheduled frequency may need adjustment if it turns out there are other NOEXERT-ignoring sources of Exhaustion, or if it turns out that properly tireless Intelligent Undead dwarves never switch away from unscheduled ICD. NOEXERT Dwarves performing scheduled training have been observed switching from ICD to sparring or demonstration.

### CODE REVIEW

**Comment by terribleperson** [comment]
pre-commit.ci autofix

File: internal/control-panel/registry.lua [hunk 1]
```diff
@@ -77,6 +77,9 @@ COMMANDS_BY_IDX = {
         params={'--time', '1', '--timeUnits', 'days', '--command', '[', 'fix/stuck-instruments', ']'}},
     {command='fix/stuck-worship', group='bugfix', mode='repeat', default=true,
         params={'--time', '1', '--timeUnits', 'days', '--command', '[', 'fix/stuck-worship', ']'}},
+    {command='fix/zero-noexert-exhaustion', group='bugfix', mode='repeat', default=false,
+        desc='Prevent NOEXERT units (notably Necromancers, Vampires, and Intelligent Undead) from becoming Tired, and fix NOEXERT units which are Tired.',
```
Comment (myk002) [inline internal/control-panel/registry.lua:81]: Try to match the description length of the other tools. The UI space for displaying this string is limited.

File: internal/control-panel/registry.lua [hunk 2]
```diff
@@ -77,6 +77,9 @@ COMMANDS_BY_IDX = {
...
+        params={'--time', '350', '--timeUnits', 'ticks', '--command', '[', 'fix/zero-noexert-exhaustion', ']'}},
```
Comment (myk002) [inline internal/control-panel/registry.lua:82]: we have up to 450 ticks to run, so 350 might be a bit too much buffer. How about 439? It's closer to the limit and is prime, so there won't be resonance cycles with other repeat tools.

File: docs/fix/zero-noexert-exhaustion.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
+fix/zero-noexert-exhaustion
```
[the remark this answers is not in the document]
Comment (terribleperson) [reply docs/fix/zero-noexert-exhaustion.rst]: Zero was in the sense of 'set the exhaustion value to zero' but I'm not averse to a shorter name and aware it looked a bit odd.

File: docs/fix/zero-noexert-exhaustion.rst [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+===========================
+
+.. dfhack-tool::
+    :summary: Sets Exhaustion counter of NOEXERT units (notably Necromancers, Vampires, and Intelligent Undead) to 0, fixing any that are stuck 'Tired'.
```
[the remark this answers is not in the document]
Comment (terribleperson) [reply docs/fix/zero-noexert-exhaustion.rst]: Fixed. I was hoping for a bit more clarity in the short summary, but no matter how I massaged the words I couldn't find a way to be more clear than "NOEXERT units' in 53 characters.

File: docs/fix/zero-noexert-exhaustion.rst [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    :tags: fort bugfix units
+
+Zeroes the Exhaustion counter of all NOEXERT units (including Vampires, Necromacers, and Intelligent Undead),
+fixing any that are stuck 'Tired' from an activity that doesn't respect NOEXERT. This is not a permanent fix -
+the issue will reoccur next time they partake in an activity that does not respect the NOEXERT tag.
+
+Running this regularly works around :bug:`8389`, which permanently debuffs NOEXERT units and prevents them from
+properly partaking in military training. It should be run if you notice Vampires, Necromancers, or Intelligent
+Undead become 'Tired'. Enabling this script via control-panel or gui/control-panel will run it often enough to
```
[the remark this answers is not in the document]
Comment (terribleperson) [reply docs/fix/zero-noexert-exhaustion.rst]: Fixed, I missed that in the doc I was referencing.

File: internal/control-panel/registry.lua (hunk 1 above)
Comment (terribleperson) [reply internal/control-panel/registry.lua]: Done, shortened description.

File: internal/control-panel/registry.lua (hunk 2 above)
Comment (terribleperson) [reply internal/control-panel/registry.lua:82]: I've seen Exhaustion reach as high as 1900 with the script on, so I don't think there's much headroom.

File: docs/fix/noexert-exhaustion.rst [hunk 6]
```diff
@@ -0,0 +1,29 @@
+fix/noexert-exhaustion
+===========================
+
+.. dfhack-tool::
+    :summary: Sets Exhaustion of NOEXERT units to 0, curing 'Tired'
```
Comment (myk002) [inline docs/fix/noexert-exhaustion.rst:5]:
```suggestion
    :summary: Prevents units with the NOEXERT tag from getting tired when training.
```

File: docs/fix/noexert-exhaustion.rst [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    :tags: fort bugfix units
+
+Zeroes the Exhaustion counter of all NOEXERT units (including Vampires, Necromacers, and Intelligent Undead),
```
Comment (myk002) [inline docs/fix/noexert-exhaustion.rst:8]:
```suggestion
This tool zeroes the "exhaustion" counter of all NOEXERT units (e.g. Vampires, Necromancers, and Intelligent Undead),
```

File: docs/fix/noexert-exhaustion.rst [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+fixing any that are stuck 'Tired' from an activity that doesn't respect NOEXERT. This is not a permanent fix -
+the issue will reoccur next time they partake in an activity that does not respect the NOEXERT tag.
+
+Running this regularly works around :bug:`8389`, which permanently debuffs NOEXERT units and prevents them from
+properly partaking in military training. It should be run if you notice Vampires, Necromancers, or Intelligent
+Undead become 'Tired'. Enabling this script via `control-panel` or `gui/control-panel` will run it often enough to
```
Comment (myk002) [inline docs/fix/noexert-exhaustion.rst:14]:
```suggestion
Undead becoming 'Tired'. Enabling this script via `control-panel` or `gui/control-panel` will run it often enough to
```

File: internal/control-panel/registry.lua [hunk 9]
```diff
@@ -77,6 +77,9 @@ COMMANDS_BY_IDX = {
         params={'--time', '1', '--timeUnits', 'days', '--command', '[', 'fix/stuck-instruments', ']'}},
     {command='fix/stuck-worship', group='bugfix', mode='repeat', default=true,
         params={'--time', '1', '--timeUnits', 'days', '--command', '[', 'fix/stuck-worship', ']'}},
+    {command='fix/noexert-exhaustion', group='bugfix', mode='repeat', default=true,
+        desc='Prevent NOEXERT units from becoming Tired',
```
Comment (myk002) [inline internal/control-panel/registry.lua:81]:
```suggestion
```

Comment (myk002) [reply internal/control-panel/registry.lua:81]: control panel picks up the short description from the rst help file by default, which is fine here.

File: docs/fix/noexert-exhaustion.rst [hunk 10]
```diff
@@ -0,0 +1,29 @@
+fix/noexert-exhaustion
+===========================
```
Comment (myk002) [inline docs/fix/noexert-exhaustion.rst:2]:
```suggestion
fix/noexert-exhaustion
======================
```

File: docs/fix/noexert-exhaustion.rst (hunk 6 above)
Comment (myk002) [reply docs/fix/noexert-exhaustion.rst:5]: 56 chars, but that is close enough

**Pull request merged**